pkg/store/limiter.go

```diff
@@ -117,16 +117,16 @@ var _ storepb.StoreServer = &limitedStoreServer{}
 type limitedStoreServer struct {
 	storepb.StoreServer
 	newSeriesLimiter      SeriesLimiterFactory
-	newSamplesLimiter     ChunksLimiterFactory
+	newChunksLimiter      ChunksLimiterFactory
 	failedRequestsCounter *prometheus.CounterVec
 }
 
 // NewLimitedStoreServer creates a new limitedStoreServer.
 func NewLimitedStoreServer(store storepb.StoreServer, reg prometheus.Registerer, selectLimits SeriesSelectLimits) storepb.StoreServer {
 	return &limitedStoreServer{
-		StoreServer:       store,
-		newSeriesLimiter:  NewSeriesLimiterFactory(selectLimits.SeriesPerRequest),
-		newSamplesLimiter: NewChunksLimiterFactory(selectLimits.SamplesPerRequest),
+		StoreServer:      store,
+		newSeriesLimiter: NewSeriesLimiterFactory(selectLimits.SeriesPerRequest),
+		newChunksLimiter: NewChunksLimiterFactory(selectLimits.SamplesPerRequest / MaxSamplesPerChunk), // The samples limit is an approximation based on the max number of samples per chunk.
```

Comment:
I don't think this is correct, because in the `Send` response we already multiple the number of chunks with 120.

Reply:
We need to unify this. We can't have `NewChunksLimiterFactory` be used so differently in two places (the other one being https://github.com/thanos-io/thanos/blob/main/cmd/thanos/store.go#L400).

Reply:
Sure, but then also change the way the limit is applied here https://github.com/thanos-io/thanos/pull/6882/files#diff-7770716fea00a6ffe0a9fb58eb95c4be170a5b7a40c929b2caa262dda555026aR174.

Reply:
Thanks for pointing it out! Missed that. 
We should probably add an e2e test for this at some point, so we can catch things like this. 🙂

Reply:
Good to know we apply store API limits not only on Store Gateway but on other components as well

Reply:
@saswatamcode I think there's a unit test for this which should have failed, but not 100% sure.